packages/synapse-react-client/src/components/SynapseTable/table-top/DownloadOptions.integration.test.tsx

```diff
@@ -0,0 +1,262 @@
+  )
+
+describe('Download Options tests', () => {
+  beforeAll(() => server.listen())
+  beforeEach(() => {
+    server.use(...getHandlersForTableQuery(mockQueryResponseData))
+  })
+  afterEach(() => server.restoreHandlers())
+  afterAll(() => server.close())
+
+  it('Shows correct options for a TableEntity with no ENTITY_ID column', async () => {
+    server.use(
+      ...getHandlersForTableQuery(mockQueryResponseDataWithoutEntityIdColumn),
+    )
+    renderComponent(props, tableEntityQueryRequest)
+
+    const downloadOptionsButton = await screen.findByRole('button', {
+      name: 'Download Options',
+    })
+
+    await userEvent.click(downloadOptionsButton)
+
+    await screen.findByRole('button', { name: 'Export Table' })
+    const programmaticOptionsMenuItem = await screen.findByRole('button', {
+      name: 'Programmatic Options',
+    })
+
+    // Download files should not be visible for tables
+    await waitFor(() => {
+      expect(
+        screen.queryByRole('button', { name: ADD_ALL_FILES_TO_DOWNLOAD_CART }),
+      ).not.toBeInTheDocument()
+    })
+    // Verify that programmatic download isn't disabled
+    expect(programmaticOptionsMenuItem.classList.contains('disabled')).toBe(
+      false,
+    )
+  })
+
+  it('Shows correct options for a File View', async () => {
+    renderComponent(props, fileViewQueryRequest)
+
+    const downloadOptionsButton = await screen.findByRole('button', {
+      name: 'Download Options',
+    })
+
+    await userEvent.click(downloadOptionsButton)
+
+    await screen.findByRole('button', { name: 'Export Table' })
+    const downloadFilesMenuItem = await screen.findByRole('button', {
+      name: ADD_ALL_FILES_TO_DOWNLOAD_CART,
+    })
+    const programmaticOptionsMenuItem = await screen.findByRole('button', {
+      name: 'Programmatic Options',
+    })
+
+    // Verify that programmatic download and download files are not disabled
+    expect(downloadFilesMenuItem.classList.contains('disabled')).toBe(false)
+    expect(programmaticOptionsMenuItem.classList.contains('disabled')).toBe(
+      false,
+    )
+  })
+
+  it('Shows correct options for a Project View', async () => {
+    renderComponent(props, projectViewQueryRequest)
+
+    const downloadOptionsButton = await screen.findByRole('button', {
+      name: 'Download Options',
+    })
+
+    await userEvent.click(downloadOptionsButton)
+
+    await screen.findByRole('button', { name: 'Export Table' })
+    const programmaticOptionsMenuItem = await screen.findByRole('button', {
+      name: 'Programmatic Options',
+    })
+
+    // Download files should not be visible for project views
+    await waitFor(() => {
+      expect(
+        screen.queryByRole('button', { name: ADD_ALL_FILES_TO_DOWNLOAD_CART }),
+      ).not.toBeInTheDocument()
+    })
+
+    // Verify that programmatic download isn't disabled
+    expect(programmaticOptionsMenuItem.classList.contains('disabled')).toBe(
+      false,
+    )
+  })
```

Comment:
By turning this into an integration test (fetching a full table) I discovered and fixed a new bug caused by the new download cart feature -- Project Views should probably never show "add to download cart", but they do before this PR

Reply:
Nice catch!